backend/lib/endpoints/organizations.js

```diff
@@ -0,0 +1,95 @@
+const httpErrors = require("http-errors");
+
+const {
+  createOrganizationSchema,
+  deleteOrganizationSchema,
+  getOrganizationSchema,
+  getOrganizationsSchema,
+  updateOrganizationSchema,
+} = require("./schema/organizations");
+
+/*
+ * /api/organizations
+ */
+async function routes(app) {
+  const Organization = app.mongo.model("Organization");
+
+  app.delete(
+    "/:organizationId",
+    {
+      preValidation: [app.authenticate],
+      schema: deleteOrganizationSchema,
+    },
+    async (req) => {
+      // TODO: make sure user can only delete organizations they own
+      return Organization.findByIdAndRemove(req.params.organizationId);
+    },
+  );
+
+  app.get(
+    "/",
+    {
+      preValidation: [app.authenticate],
+      schema: getOrganizationsSchema,
+    },
+    async (req, reply) => {
+      const { ownerId } = req.params;
+      const filter = ownerId ? { ownerId } : {};
+      const sortedOrganizations = await Organization.find(filter).sort({
+        name: 1,
+      });
+      return reply.send(sortedOrganizations);
+    },
+  );
+
+  app.get(
+    "/:organizationId",
+    {
+      preValidation: [app.authenticate],
+      schema: getOrganizationSchema,
+    },
+    async (req) => {
+      const result = await Organization.findById(req.params.organizationId);
+      if (result === null) {
+        return new httpErrors.NotFound();
+      }
+      return result;
+    },
+  );
+
+  app.patch(
+    "/:organizationId",
+    {
+      preValidation: [app.authenticate],
+      schema: updateOrganizationSchema,
+    },
+    async (req, reply) => {
+      // TODO: make sure user can only update organizations they own
+      const organization = await Organization.findById(
+        req.params.organizationId,
+      );
+      if (organization === null) {
+        return reply.send(new httpErrors.NotFound());
+      }
+      Object.keys(req.body).forEach((key) => {
+        if (organization[key] !== req.body[key]) {
```

Comment:
what do you think about also checking if `organization[key]` exists? So something like: 

`if (organization[key] && organization[key] !== req.body[key]) `

though now that I think of it, it might not be a huge difference

Reply:
@anniegiang actually, I think the check is pretty important since we're directly using client's request body which may have more number of keys (like a malicious attempt) which will cause server error.

Reply:
Also, I was thinking instead of directly assigning the request body keys to the model keys, it's better to first filter out only the desired keys by maybe validating the schema.
Otherwise, what if a malicious user changes a private (like maybe created timestamp) field of the Organization object?

Reply:
So, I originally had the check for `organization[key]` but the way MongoDB works is that if you don't set some of the fields, the keys won't even exist in the record. So for instance, if you do a POST with the following body (shortened for clarity):

```
{
  "name": "Test org",
  "email": "test@example.com"
}
```

And then later you do a PATCH to set another field:
```
{
  "description": "An awesome organization"
}
```

When we fetch the organization record from MongoDB, the `description` key won't exist, so it will be skipped, and consequently won't be set.

Reply:
I do like the idea of whitelisting fields that can be updated, so I can add that.

Reply:
Another thing I was thinking about - should we set any unset fields coming into a POST create request to `null` so that there is at least a key set on the MongoDB object?

Reply:
@robinv85 @anniegiang @shivanshs9 I decided to check the properties defined on `updateOrganizationSchema` instead of on the `organization` object returned from MongoDB. As I mentioned, the `organization` object may not have all fields set, so patching of any missing fields will not work. I also added error checking so that we can return a 400 bad request error if any extra fields are passed in through the patch request. See 34a650c

Regarding my comment about system fields, if we do decide to add them, we can do that in a separate ticket/PR as it's kind of outside the scope of this ticket.